File: core_program/peak_detection.py
import time
import numpy as np
import numexpr as ne

if __name__ == "__main__":
    start_time = time.time()
    num1 = [i for i in range(10000)] #speed test
    num2 = [i for i in range(10000)] #speed test
    res = 0
    for x in num1:
        for y in num2:
            res = res + x + y
    print(res)

    all_time = time.time() - start_time
    print("Execution time:{0} [sec]".format(all_time))

File: core_program/peak_detection.c
#include "peak_detection.h"

int c_peak_detect(int *num1, int *num2, int size_a, int size_b){
    int res = 0;
    for(int i=0; i < size_a; i++){
        for(int j=0; j < size_b; j++){
            res = res + num1[i]+num2[j];
        }
    }
    return res;
}
